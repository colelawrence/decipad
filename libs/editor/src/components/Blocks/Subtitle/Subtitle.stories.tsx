import { Subtitle } from './Subtitle.component';

export default {
  title: 'Editor/Blocks/Subtitle',
  component: Subtitle,
};

export const Default = (args: { text: string }) => (
  <Subtitle
    leaf={{ attributes: {}, text: '' }}
    element={{ children: [{ text: '' }] }}
    text={{ text: '' }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
      'data-slate-leaf': true,
    }}
  >
    {args.text}
  </Subtitle>
);

Default.args = {
  text: 'Subtitle',
};

Default.storyName = 'Subtitle';
