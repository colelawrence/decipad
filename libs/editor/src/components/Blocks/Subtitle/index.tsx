import styled from '@emotion/styled';
import { SlatePluginComponent } from '@udecode/slate-plugins-core';
import React from 'react';

const Element = styled('h2')({
  fontSize: '24px',
  margin: 0,
  padding: '2rem 0 1rem 0',
  fontWeight: 'bold',
  color: '#121214',
});

export const Subtitle: SlatePluginComponent = ({ attributes, children }) => {
  return <Element {...attributes}>{children}</Element>;
};
