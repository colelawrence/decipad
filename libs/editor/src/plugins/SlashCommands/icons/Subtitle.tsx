import { FC } from 'react';

export const Subtitle = (): ReturnType<FC> => {
  return (
    <svg
      width="14"
      height="14"
      viewBox="0 0 14 14"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.1871 0.710938H0.812109C0.648007 0.710939 0.490627 0.776129 0.374589 0.892167C0.258551 1.0082 0.193361 1.16559 0.193359 1.32969V3.80469C0.193359 3.96879 0.258549 4.12617 0.374587 4.24221C0.490625 4.35825 0.648007 4.42344 0.812109 4.42344C0.976212 4.42344 1.13359 4.35825 1.24963 4.24221C1.36567 4.12617 1.43086 3.96879 1.43086 3.80469V1.94844H6.38086V11.8484H4.52461C4.36051 11.8484 4.20313 11.9136 4.08709 12.0297C3.97105 12.1457 3.90586 12.3031 3.90586 12.4672C3.90586 12.6313 3.97105 12.7887 4.08709 12.9047C4.20313 13.0207 4.36051 13.0859 4.52461 13.0859H9.47461C9.63871 13.0859 9.79609 13.0207 9.91213 12.9047C10.0282 12.7887 10.0934 12.6313 10.0934 12.4672C10.0934 12.3031 10.0282 12.1457 9.91213 12.0297C9.79609 11.9136 9.63871 11.8484 9.47461 11.8484H7.61836V1.94844H12.5684V3.80469C12.5684 3.96879 12.6335 4.12617 12.7496 4.24221C12.8656 4.35825 13.023 4.42344 13.1871 4.42344C13.3512 4.42344 13.5086 4.35825 13.6246 4.24221C13.7407 4.12617 13.8059 3.96879 13.8059 3.80469V1.32969C13.8059 1.16559 13.7407 1.0082 13.6246 0.892167C13.5086 0.776129 13.3512 0.710939 13.1871 0.710938Z"
        fill="#121214"
      />
    </svg>
  );
};
