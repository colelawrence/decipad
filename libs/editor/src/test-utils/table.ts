import {
  ELEMENT_TABLE,
  ELEMENT_TH,
  ELEMENT_TR,
  ELEMENT_TD,
} from '@udecode/plate';
import {
  InteractiveTable,
  TableCellType,
  Td,
  Th,
} from '../plugins/InteractiveTable/table';
import {
  ELEMENT_TABLE_CAPTION,
  ELEMENT_THEAD,
  ELEMENT_HEAD_TR,
  ELEMENT_TBODY,
} from '../utils/elementTypes';

/**
 * Creates a row with 1 named column and {@param numberOfRows} rows of non-empty cells.
 */
export const createTable = (numberOfRows: number): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType: 'string' },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: Array.from({ length: numberOfRows }, (_, i) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType: 'string' },
            children: [{ text: `Col 1 Row ${i + 1}` }],
          },
        ],
      })),
    },
  ],
});

export const createTableWithTypeAndData = (
  cellType: TableCellType,
  data: string[]
): InteractiveTable => ({
  type: ELEMENT_TABLE,
  children: [
    {
      type: ELEMENT_TABLE_CAPTION,
      children: [{ text: 'Table' }],
    },
    {
      type: ELEMENT_THEAD,
      children: [
        {
          type: ELEMENT_HEAD_TR,
          children: [
            {
              type: ELEMENT_TH,
              attributes: { cellType },
              children: [{ text: 'Col 1' }],
            },
          ],
        },
      ],
    },
    {
      type: ELEMENT_TBODY,
      children: data.map((text) => ({
        type: ELEMENT_TR,
        children: [
          {
            type: ELEMENT_TD,
            attributes: { cellType },
            children: [{ text }],
          },
        ],
      })),
    },
  ],
});

export const printTable = ({
  children: [_, thead, tbody],
}: InteractiveTable): string[] => {
  const tdsToStrings = (tds: (Th | Td)[]) =>
    tds.map((td) => td.children[0].text).join(' | ');

  return [
    tdsToStrings(thead.children[0].children),
    ...tbody.children.map((tr) => tdsToStrings(tr.children)),
  ];
};
