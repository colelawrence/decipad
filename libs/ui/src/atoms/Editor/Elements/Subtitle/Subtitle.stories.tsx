import { SubtitleElement } from './Subtitle';

export default {
  title: 'Atoms/Editor/Elements/Subtitle',
  component: SubtitleElement,
  args: { children: 'Subtitle Element' },
};

interface ArgsType {
  children: string;
}

export const Subtitle = (args: ArgsType) => (
  <SubtitleElement
    attributes={{ 'data-slate-leaf': true, 'data-slate-node': 'element' }}
    leaf={{ text: '' }}
    text={{ text: '' }}
    nodeProps={{ styles: { root: { css: null } } }}
  >
    {args.children}
  </SubtitleElement>
);
