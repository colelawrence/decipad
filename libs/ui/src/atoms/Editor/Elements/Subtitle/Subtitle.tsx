import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import { h1 } from '../../../../primitives';

const styles = css(h1, {
  padding: '16px 0',
});

export const SubtitleElement: PlatePluginComponent = ({
  attributes,
  children,
}) => {
  return (
    <h2 css={styles} {...attributes}>
      {children}
    </h2>
  );
};
