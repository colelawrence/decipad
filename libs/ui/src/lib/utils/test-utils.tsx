import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

import { Type } from '@decipad/language';
import { theme, ResultsContextProvider } from '../..';

const results = {
  test: {
    type: Type.build({
      type: 'number',
      unit: {
        unit: 'apples',
      },
    } as any),
    value: [20],
    errors: [],
  },
  noResult: null,
};

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <ResultsContextProvider value={results}>
        {children}
      </ResultsContextProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
