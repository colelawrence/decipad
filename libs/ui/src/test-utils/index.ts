export * from './menu';
export * from './language';
