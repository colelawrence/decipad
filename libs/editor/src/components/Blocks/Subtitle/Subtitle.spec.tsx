import { render } from 'test-utils';
import { Default } from './Subtitle.stories';

describe('Subtitle Block', () => {
  it('matches snapshot', () => {
    const { container } = render(<Default text="Subtitle" />);
    expect(container).toMatchSnapshot();
  });
});
