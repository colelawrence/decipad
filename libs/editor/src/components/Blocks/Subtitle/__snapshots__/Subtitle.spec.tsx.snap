// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle Block matches snapshot 1`] = `
.emotion-0 {
  font-size: 24px;
  margin: 0;
  padding: 2rem 0 1rem 0;
  font-weight: bold;
  color: #121214;
}

.emotion-0 ::selection {
  background-color: rgba(196, 202, 251, 0.5);
}

<div>
  <h2
    class="emotion-0"
    data-slate-leaf="true"
    data-slate-node="element"
  >
    Subtitle
  </h2>
</div>
`;
